programs/ziti-edge-tunnel/ziti-edge-tunnel.c

```diff
@@ -2386,92 +2344,98 @@ static int dump_opts(int argc, char *argv[]) {
     return optind;
 }
 
-static void on_response(uv_stream_t *s, ssize_t len, const uv_buf_t *b) {
-    if (len > 0) {
-        printf("received response <%.*s>\n", (int) len, b->base);
-    } else {
-        fprintf(stderr,"Read Response error %s\n", uv_err_name(len));
-    }
-    uv_read_stop(s);
-    free(b->base);
-    uv_close((uv_handle_t *)s, NULL);
-}
-
-void on_write(uv_write_t* req, int status) {
-    if (status < 0) {
-        fprintf(stderr,"Could not sent message to the tunnel. Write error %s\n", uv_err_name(status));
+static int send_message_to_tunnel(char* message, bool show_result) {
+#if _WIN32
+    HANDLE cmd_soc = CreateFileA(sockfile,
+                                 GENERIC_READ | GENERIC_WRITE,
+                                 0, NULL,
+                                 OPEN_EXISTING,
+                                 FILE_FLAG_OVERLAPPED, NULL);
+    if (cmd_soc == INVALID_HANDLE_VALUE) {
+        DWORD err = GetLastError();
+        fprintf(stderr, "failed to connect to pipe: %lu", err);
+        exit(1);
```

Comment:
are you certain about this particular change with respect to `exit`-ing? Why is windows the only one that would end up in this situation? Shouldn't the other OS's all exit too in these situations?

This function is much harder to review with all these defines. the old code appears much simpler (just looking at it here, without the IDE) why were these changes needed?

Reply:
other OS's would also exit -- just a bit later -- after the write attempt. I'll add the exit on connect

there were multiple callback needed for a simple request/response pattern. which is definitely not simpler

also almost all `ziti-edge-tunnel <subcmd>` were not working on windows because command parsing was relying on the `'\0'/'\n'`